Add Chrome tests to bazel build.

TriciaCrichton opened the pull request:
<!-- NOTE
Please be aware that the Selenium Grid 3.x is being deprecated in favour of the
upcoming version 4.x. We won't be receiving any PRs related to the Grid 3.x code. Thanks!
-->
Added BUILD.bazel file for the java Chrome tests. This is modeled after the devtools build file.



- [X] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

TriciaCrichton commented:
The first failures in the build are from running FormHandlingTest for Firefox. That is not related to this change.

The second failures are caused because file needed by ChromeOptionsFunctionalTest is not found. The test was added in 2017; the file stil exists in third_party/chrome_ext. I think we just need to update the path. How do we point to the third_party directory for the bazel build?

TriciaCrichton commented:
fixes #7785

barancev commented:
We've changed Java build infrastructure to use external dependencies instead of stored in third_party directory:
https://github.com/SeleniumHQ/selenium/commit/6085af940e00132ef09b8594876ae5240575f520

Can you please update this PR to use external dependencies too.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SeleniumHQ/selenium?pullRequest=7756) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/SeleniumHQ/selenium?pullRequest=7756) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SeleniumHQ/selenium?pullRequest=7756) it.</sub>

TriciaCrichton commented:
Updated the build file to use external dependencies.
Because I don't know how to setup access to third_party/chrome_ext, I disabled that test.

We use the ChromeDriverTests file to run the tests locally, so I replaced the imports of other *Tests files with explicit lists of the included tests.

TriciaCrichton commented:
There are 3 unrelated test failures in the build: 
FormHandlingTest.testShouldSubmitAFormUsingTheNewlineLiteral - broken by a change to the Selenium code, see issue 7886
PageLoadingTest.testShouldBeAbleToAccessPagesWithAnInsecureSslCertificate - I submitted PR 7888 to fix the test.
ChromeDevToolsNetworkTest.verifyCertificatesAndOverrideUserAgent - the test does not configure ChromeDriver to Accept Insecure Certificates. This isn't a change I feel confident in making.

TriciaCrichton commented:
@barancev Can you take a look at this again? I changed to use external dependencies.

shs96c commented on java/client/test/org/openqa/selenium/chrome/ChromeDriverTests.java:
```diff
@@ -19,15 +19,104 @@
-}
+  /* Use a custom suite to enable use of Class.forName, which is not allowed
+      by the annotations. This isolates tests run by ChromeDriver from changes
+      in the build tools. */ 
```
We're not planning on shifting build tools again for a while, so I'm not sure that this is a concern.

shs96c commented on java/client/test/org/openqa/selenium/chrome/ChromeDriverTests.java:
```diff
@@ -19,15 +19,104 @@
-    DevToolsTests.class
-})
+@RunWith(ChromeDriverTests.ChromeTestRunner.class)
 public class ChromeDriverTests {
```
The way we're working on making this work is to avoid test suites, and instead use `java_selenium_test_suite` in our `BUILD.bazel` files. This creates a target per test, and so allows us to shard them more efficiently. It's defined in `//java:defs.bzl`.

That takes a `browser` attribute: https://github.com/SeleniumHQ/selenium/blob/master/java/private/test.bzl#L29 which I suspect you want to set to `["chrome", "edge"]`

diemol commented:
Closing as there was a follow up at #8275
